src/werkzeug/debug/shared/debugger.js

```diff
@@ -1,166 +1,255 @@
+            const img = createIconForConsole();
+            img.addEventListener('click', (e) => {
+                e.stopPropagation();
+                consoleNode = openShell(consoleNode, target, frameID);
+                return false;
+            });
+            target.getElementsByTagName("pre")[j].append(img);
+        }
+    }
+}
+
+function slideToggle(target) {
+    target.classList.toggle('active');
+}
 
 /**
- * Helper function for shell initialization
+ * toggle traceback types on click.
  */
-function openShell(consoleNode, target, frameID) {
-  promptForPin();
-  if (consoleNode)
-    return consoleNode.slideToggle('fast');
-  consoleNode = $('<pre class="console">')
-    .appendTo(target.parent())
-    .hide()
-  var historyPos = 0, history = [''];
-  var output = $('<div class="output">[console ready]</div>')
-    .appendTo(consoleNode);
-  var form = $('<form>&gt;&gt;&gt; </form>')
-    .submit(function() {
-      var cmd = command.val();
-      $.get('', {
-          __debugger__: 'yes', cmd: cmd, frm: frameID, s: SECRET}, function(data) {
-        var tmp = $('<div>').html(data);
+function addToggleTraceTypesOnClick(elements) {
+    for (let i = 0; i < elements.length; i++) {
+        elements[i].addEventListener('click', () => {
+            $(this).next().slideToggle('fast');
+            $('div.plain').slideToggle('fast');
+        });
+        elements[i].style.cursor = 'pointer';
+        document.querySelector('div.plain').style.display = 'none';
+    }
+}
+
+function plainTraceback(elements) {
+  elements.replaceWith($('<pre>').text(plainTraceback.text()))
+}
+
+function createConsole() {
+  const consoleNode = document.createElement('pre');
+  consoleNode.classList.add("console");
+  consoleNode.classList.add("active");
+  return consoleNode;
+}
+
+function createConsoleOutput() {
+  const output = document.createElement('div');
+  output.classList.add('output');
+  output.innerHTML = '[console ready]';
+  return output;
+}
+
+function createConsoleInputForm() {
+  const form = document.createElement('form');
+  form.innerHTML = '&gt;&gt;&gt; ';
+  return form;
+}
+
+function createConsoleInput() {
+  const command = document.createElement("input");
+  command.type = "text";
+  command.setAttribute("autocomplete", "off");
+  command.setAttribute("spellcheck", false);
+  command.setAttribute("autocapitalize", "off");
+  command.setAttribute("autocorrect", "off");
+  return command;
+}
+
+function createIconForConsole() {
+  let img = document.createElement('img');
+  img.setAttribute("src", "?__debugger__=yes&cmd=resource&f=console.png");
+  img.setAttribute('title', 'Open an interactive python shell in this frame');
+  return img;
+}
+
+function handleConsoleSubmit(e, command, frameID) {
+  // Prevent page from refreshing.
+  e.preventDefault();
+
+  return new Promise((resolve, reject) => {
+    // Get input command.
+    let cmd = command.value;
+
+    // Setup GET request.
+    let http = new XMLHttpRequest();
```

Comment:
What about using `fetch`? Much less boilerplate than `XMLHttpRequest`, and natively support in all relevant browsers.